internal/dcache/rpc/client/client_pool.go

```diff
@@ -1091,6 +1112,67 @@ func (cp *clientPool) deleteNodeClientPoolIfInactive(nodeID string) bool {
 	return true
 }
 
+// AddIffyRV adds an RV to the iffyRVMap.
+// When PutChunkDC() fails with timeout error, we add the next-hop RV and the next RVs in chain
+// to the iffyRVMap.
+func (cp *clientPool) addIffyRV(rvName string) {
+	common.Assert(cm.IsValidRVName(rvName), rvName)
+	cp.iffyRVMap.Store(rvName, time.Now())
+	log.Debug("clientPool::addIffyRV: added %s to IffyRVMap at %v", rvName, time.Now())
+}
+
+// RemoveIffyRV removes an RV from the iffyRVMap.
+// An RV is removed from the map by,
+//   - periodicRemoveIffyRVs() goroutine which checks if the iffyRVTimeout has expired for the RV.
+//   - successful RPC call to the RV indicating that the connection between the client and the
+//     RV is healthy.
+func (cp *clientPool) removeIffyRV(rvName string) {
+	common.Assert(cm.IsValidRVName(rvName), rvName)
+
+	if !cp.isIffyRV(rvName) {
+		return
+	}
+
+	cp.iffyRVMap.Delete(rvName)
+	log.Debug("clientPool::removeIffyRV: removed %s from IffyRVMap", rvName)
+}
+
+// Check if an RV is marked iffy.
+func (cp *clientPool) isIffyRV(rvName string) bool {
+	common.Assert(cm.IsValidRVName(rvName), rvName)
+
+	_, ok := cp.iffyRVMap.Load(rvName)
+	return ok
+}
+
+// Goroutine which runs every iffyRVTimeout and removes expired RVs from the iffyRVMap.
+func (cp *clientPool) periodicRemoveIffyRVs() {
+	for {
+		select {
+		case <-cp.iffyDone:
+			log.Info("clientPool::periodicRemoveIffyRVs: stopping periodic removal of iffy RVs")
+			return
+		case <-cp.iffyTicker.C:
+			cp.iffyRVMap.Range(func(key, value any) bool {
+				rvName, ok := key.(string)
+				_ = ok
+				common.Assert(ok, key)
+				common.Assert(cm.IsValidRVName(rvName), rvName)
+
+				addedTime, ok := value.(time.Time)
+				_ = ok
+				common.Assert(ok, addedTime)
+
+				if time.Since(addedTime) > defaultIffyRVTimeout {
+					cp.removeIffyRV(rvName)
```

Comment:
add one more log here so that we know it's remove from here and not the successful rpc case

Reply:
Done, added